Utils/MiscUtils.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace ThunderstoreCLI
+{
+    public static class MiscUtils
+    {
+        /// <summary>Return application version</summary>
+        /// Version number is controlled via MinVer by creating new tags
+        /// in git. See README for more information.
+        public static int[] GetCurrentVersion()
+        {
+            string version;
+
+            try
+            {
+                version = Assembly.GetEntryAssembly()!
+                    .GetCustomAttribute<AssemblyInformationalVersionAttribute>()!
+                    .InformationalVersion;
+            }
+            catch (NullReferenceException)
+            {
+                throw new Exception("Reading app version from assembly failed");
+            }
+
+            // Drop possible pre-release cruft ("-alpha.0.1") from the end.
+            var versionParts = version.Split('-')[0].Split('.');
+
+            if (versionParts is null || versionParts.Length != 3)
+            {
+                throw new Exception("Malformed app version: ${version}");
+            }
+
+            return versionParts.Select(part => Int32.Parse(part)).ToArray();
+        }
+
+        /// <summary>Extract version from release information</summary>
+        /// <exception cref="ArgumentException">Throw if version number not found</exception>
+        public static int[] ParseLatestVersion(string releaseJsonData)
+        {
+            var regex = new Regex(@"""tag_name"":""(\d+.\d+.\d+)""");
```

Comment:
Do we need full semver support here now, or is the suffix (e.g. `-alpha`) not included in the binary?

Reply:
`GetCurrentVersion` reads the value from binary. This method reads the value from JSON metadata returned by GitHub API. My idea was that there would be no notifications about alpha-releases, i.e. this would throw the `ArgumentException` and the caller method would fail silently.

But now that I think about this, this would also fail silently is situations where user has e.g. `0.1.0`, there's a stable `0.2.0` release, but the latest one is `0.3.0-alpha.1`.

So I guess that if we want to support alpha-releases (instead of just tags), this needs to be switched so that we fetch information about more than just the latest release. It would also be required if there's going to be e,g, more than one supported major release at a time (e.g. releasing `1.2.3` and `2.0.2` that contain the same hotfix at the same time).

Would it be a good idea to support those cases at this time?

Reply:
I'd say if it's not too much work (1d tops), then sure, would hate the update notification breaking entirely because someone pushes a suffixed version. But it's also not an irrecoverable state, as you can just push a new release with a proper tag format.

https://github.com/maxhauser/semver this is the closest parser I found with a quick search, but it also seems rather unmaintained. Still could be good for pointers at least?

Reply:
Well I think we just need to query a different endpoint, and parse release tags in a loop, and then pick the latest stable release, so it shouldn't take that long. And I'll see if I can do it without external dependencies first.

I still assume we don't want to show the notification for alpha-releases.

Reply:
Sounds good to me, and yeah makes sense to only show full

Reply:
* Added a [comparer for semvers](https://github.com/thunderstore-io/thunderstore-cli/pull/24/commits/44b8d9d3809560723e8e21e0d822cb30e385485e#diff-a240d47b5472ff5ff9ca38a4665ec9cd04254a6dd79d4b30bdf1e13ceafaf6cbR1)
* Rewrote [ParseLatestVersion](https://github.com/thunderstore-io/thunderstore-cli/pull/24/commits/44b8d9d3809560723e8e21e0d822cb30e385485e#diff-1d63bd23e24df93ef887ec1a8a8d8d6907959b73e0da44556b4cda5133adaaf3R43) to process a string that contains multiple releases
* Renamed the [fetcher method](https://github.com/thunderstore-io/thunderstore-cli/pull/24/commits/44b8d9d3809560723e8e21e0d822cb30e385485e#diff-1d63bd23e24df93ef887ec1a8a8d8d6907959b73e0da44556b4cda5133adaaf3R63) and updated the endpoint URL, so now all releases are fetched, not just the latest. Also updated the place where the method is called.

Other places weren't touched.

Reply:
Thanks 👍